src/protocol/mod.rs

```diff
@@ -449,9 +455,22 @@ impl WebSocketContext {
                     } else {
                         // The server MUST close the connection upon receiving a
                         // frame that is not masked. (RFC 6455)
-                        return Err(Error::Protocol(
-                            "Received an unmasked frame from client".into(),
-                        ));
+                        // The only exception here is if the user explicitly accepts given
+                        // stream (by tungstenite::server::accept_with_config or tungstenite::server::accept_hdr_with_config)
+                        // with WebSocketConfig.server_allow_unmasked set to Some(true)
+                        if let Some(server_allow_unmasked_val) =
+                            self.get_config().server_allow_unmasked
+                        {
+                            if server_allow_unmasked_val == false {
```

Comment:
Just some Rust fu: even when server_allow_unmasked is `Option<bool>`, this wouldn't have to be 2 nested ifs - mere `if self.get_config().server_allow_unmasked == Some(true)` should have sufficed.

(this comment is void if we change server_allow_unmasked to plain bool)

Reply:
nice! i will redo based on final decision whether **_server_allow_unmasked_** should be **_bool_** or **_Option<bool>_**

Reply:
My suggestion would be `if self.get_config().server_allow_unmasked.unwrap_or(false)`. But the above one is fine too.